core/src/main/java/uk/gov/justice/services/core/handler/HandlerMethod.java

```diff
@@ -98,15 +105,14 @@ private static boolean isEnvelope(final Class<?> clazz) {
      * null {@link Void}
      */
     @SuppressWarnings("unchecked")
-    public Object execute(final JsonEnvelope envelope) {
+    public <T> Envelope<T> execute(final Envelope<T> envelope) {
```

Comment:
Input and output might not be the same type. I think you probably want wildcards here.